src/main/webapp/js/components/entities/users/user-edit.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import { Form } from 'react-final-form';
+import Alert from 'react-s-alert';
+import { withRouter } from 'react-router-dom';
+
+import { getEntityMemberById } from '../../../selectors';
+import renderFormField, { validate } from '../../../utils/form/form-utils';
+import { updateEntity, deleteEntity } from '../../../actions/entity-actions';
+import ConfirmModal from '../../comfirm-modal';
+import { ADMIN } from '../../../constants/permissions';
+
+const ALERT_TIMEOUT = 5000;
+
+class UserEdit extends Component {
```

Comment:
you should not copy the code from EntityEdit here, you can use it here. this component should be similar to EditEntityPage (the new component created from CreateOrEditEntityPage)
you will fetch and override the field config in it (set editable to false for roles and and username fields)